jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java

```diff
@@ -527,4 +541,67 @@ public TaskService initTaskService(Context<?> context, InternalRuntimeManager ma
 
     }
 
+    
+    /*
+     * locking support for same context - runtime engine that deals with exact same process instance context
+     */
+    
+    protected void createLockOnGetEngine(Context<?> context, RuntimeEngine runtime) {
+        if (!useLocking) {
+            logger.debug("Locking on runtime manager disabled");
+            return;
+        }
+        
+        if (context instanceof ProcessInstanceIdContext) {
+            Long piId = ((ProcessInstanceIdContext) context).getContextId();
+            if (piId != null) {
```

Comment:
when you start new process instance, then the context is without process instance id as it was not yet created